R/repeat_linter.R

```diff
@@ -0,0 +1,46 @@
+#' Repeat linter
+#'
+#' Check that `while` is not used for infinite loops.
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "while (TRUE) { }",
+#'   linters = repeat_linter()
+#' )
+#'
+#'
+#' # okay
+#' lint(
+#'   text = "repeat { }",
+#'   linters = repeat_linter()
+#' )
+#'
+#'
+#' @evalRd rd_tags("repeat_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+repeat_linter <- function() {
+  xpath <- "
+  //WHILE[following-sibling::*[1][self::OP-LEFT-PAREN] and
```

Comment:
first condition looks redundant to me. is it possible to write a while loop without '('? if you're worried about 'while' being used in other contexts, while is a reserved word so it will show up as a SYMBOL IINM (worth adding a test for this to be sure)

Reply:
> while is a reserved word so it will show up as a SYMBOL IINM (worth adding a test for this to be sure)
When I try to create a test for it, the parser always flags an error. I would skip the tests if you don't have a suggestion

Reply:
This works: ``parse(text = "`while`(TRUE, NULL)")``, but actually resolves as equivalent to `while (TRUE) NULL`.

We can actually get into really funky territory here:

```r
`while` <- identity
while(TRUE)
# [1] TRUE
while (FALSE) { }
# Error in while (FALSE) { : unused argument ({
# })
```

I'm OK to skip tests.